src/GovUk.Education.ExploreEducationStatistics.Data.Processor/Services/ReleaseProcessorService.cs

```diff
@@ -1,252 +0,0 @@
-                        Title = message.Release.Publication.Title,
-                        Slug = message.Release.Publication.Slug,
-                        Topic = topic
-                    });
-        }
-
-        private Task<Topic> CreateOrUpdateTopic(ImportMessage message, StatisticsDbContext statisticsDbContext,
-            Theme theme)
-        {
-            return CreateOrUpdateWithExclusiveLock(
-                message.Release.Publication.Topic.Title,
-                statisticsDbContext,
-                message.Release.Publication.Topic,
-                message.Release.Publication.Topic.Id,
-                new Topic
-                    {
-                        Id = message.Release.Publication.Topic.Id,
-                        Title = message.Release.Publication.Topic.Title,
-                        Slug = message.Release.Publication.Topic.Slug,
-                        Theme = theme
-                    });
-        }
-
-        private Task<Theme> CreateOrUpdateTheme(ImportMessage message, StatisticsDbContext statisticsDbContext)
-        {
-            return CreateOrUpdateWithExclusiveLock(
-                message.Release.Publication.Topic.Theme.Title,
-                statisticsDbContext,
-                message.Release.Publication.Topic.Theme,
-                message.Release.Publication.Topic.Theme.Id,
-                new Theme 
-                {
-                    Id = message.Release.Publication.Topic.Theme.Id,
-                    Slug = message.Release.Publication.Topic.Theme.Slug,
-                    Title = message.Release.Publication.Topic.Theme.Title
-                });
-        }
-        
-        /**
-         * This method allows for a concurrency-safe pattern to check for the existence of a given entity and, if it
-         * does not exist to create it, or if it does exist, to update it.
-         *
-         * This is concurrency-safe as it ensures that any two concurrent importer processes that are trying to create
-         * or update the same entity type with the same ID will not be able to do so at the same time.  Rather, the
-         * first thread to reach this code will acquire a lock and prevent the second thread from being able to check
-         * for the existence of the entity until the first thread has finished creating it.  At this point, the second
-         * thread is freed up to check if the entity already exists, which by this point, it will.
-         */
-        private Task<TEntity> CreateOrUpdateWithExclusiveLock<TEntity, TMessageObject>(
-            string entityName,
-            StatisticsDbContext statisticsDbContext,
-            TMessageObject objectFromMessage,
-            Guid entityId,
-            TEntity entityToCreate)
-            where TEntity : class 
-            where TMessageObject : class
-        {
-            var typeName = typeof(TEntity).Name;
-            
-            return DbUtils.ExecuteWithExclusiveLock(
```

Comment:
This leaves the method `Data.Imporer.Utils.DbUtils.ExecuteWithExclusiveLock` unused.